virtcontainers/container.go

```diff
@@ -432,6 +459,19 @@ func (c *Container) rollbackFailingContainerCreation() {
 	}
 }
 
+func (c *Container) checkBlockDeviceSupport() bool {
```

Comment:
So we are going to have different semantics with different underlying hypervisor? Might just reject the usage if the hypervisor does not support passing block devices like this, at least we are sticking with a constant API semantics.

Reply:
Different hypervisors would indicate if they support block devices using the capabilities() interface, whereas the DisableBlockDeviceUse is meant to be a user config to use block devices/ 9p.

Reply:
My point is that we need to make sure the kata library API is constant across different hypervisors. E.g., when we get the same OCI spec, we should deliver the same container fs layout, no matter which hypervisor is chosen. So if `DisableBlockDeviceUse` is set to false, and the hypervisor does not support it, we should just fail the API instead of trying to work around it.

Reply:
The approach so far has been to fallback to 9p incase the hypervisor does not support it. The block device usage is an optimization after all. You dont agee? The rootfs is also